src/entities/user/user.controller.spec.ts

```diff
@@ -0,0 +1,288 @@
+      req.body = { login: 'login', password: 'password', age: 44 };
+      const createdUser = ({
+        ...req.body,
+        id: 'some-id',
+        isDeleted: false,
+      } as unknown) as UserModel;
+
+      jest
+        .spyOn(userService, 'create')
+        .mockImplementation(() => Promise.resolve(createdUser));
+
+      await createUser(req, res);
+
+      expect(userService.create).toHaveBeenCalledWith(req.body);
+    });
+
+    it('should call res.send with Error status 500 if create user operation throws error', async () => {
+      req.body = {};
+
+      const errorMsg = 'some error';
+
+      jest
+        .spyOn(userService, 'create')
+        .mockImplementation(() => Promise.reject(new Error(errorMsg)));
+      await createUser(req, res);
+
+      expect(res.send).toHaveBeenCalledWith(errorMsg);
+      expect(res.status).toHaveBeenCalledWith(ErrorCodes.InternalServerError);
+    });
+  });
+
+  describe('updateUser', () => {
+    it('should call user service with user data', async () => {
+      req.params = {
+        userId: '1',
+      };
+      req.body = { login: 'login', password: 'password', age: 44 } as UserModel;
+
+      jest.spyOn(userService, 'update').mockImplementation(() =>
+        Promise.resolve({
+          ...req.body,
+          id: '1',
+          isDeleted: false,
+        }),
+      );
+
+      await updateUser(req, res);
+
+      expect(userService.update).toHaveBeenCalledWith(req.body, '1');
+    });
+
+    it('should return 404 if user to update was not in DB', async () => {
+      req.params = {
+        userId: '1',
+      };
+      req.body = { login: 'login', password: 'password', age: 44 } as UserModel;
+
+      jest
+        .spyOn(userService, 'update')
+        .mockImplementation(() => Promise.resolve(null));
+
+      await updateUser(req, res);
+
+      expect(res.status).toHaveBeenCalledWith(ErrorCodes.NotFound);
+      expect(res.send).toHaveBeenCalledTimes(1);
+    });
+
+    it('should call res.send with Error status 500 if update user operation throws error', async () => {
+      req.params = {
+        userId: '1',
+      };
+      req.body = {};
+
+      const errorMsg = 'some error';
+
+      jest
+        .spyOn(userService, 'update')
+        .mockImplementation(() => Promise.reject(new Error(errorMsg)));
+      await updateUser(req, res);
+
+      expect(res.send).toHaveBeenCalledWith(errorMsg);
+      expect(res.status).toHaveBeenCalledWith(ErrorCodes.InternalServerError);
+    });
+  });
+
+  describe('softDeleteUser', () => {
+    it('should call user service with user id', async () => {
+      req.params = {
+        userId: '1',
+      };
+
+      jest.spyOn(userService, 'softDelete').mockImplementation(() =>
+        Promise.resolve({
+          ...req.body,
+          id: '1',
+          isDeleted: true,
+        }),
+      );
+
+      await softDeleteUser(req, res);
+
+      expect(userService.softDelete).toHaveBeenCalledWith('1');
```

Comment:
Expect for message `"Deleted user ${userId}"` is missing.

Reply:
added